New Block: UDS Banner

wmcconnell opened the pull request:
While we have a widget, and a widget area, for banners across multiple pages - we don't have a way to easily insert a single UDS banner on a page. With the COVID banner seeming demoted to more of an 'alert' status, but still retaining banner styles, I figured we should also have a block for adding banners.

Already wrapped in a fluid container, the block can be used directly at the root level of a page for a max-width banner, like this:

<img width="399" alt="Screen Shot 2021-08-19 at 4 30 39 PM" src="https://user-images.githubusercontent.com/31013359/130157797-418453a0-84d9-4dac-bb11-4a85b844815e.png">

Or within a container/row/column structure to fit within the content area, like this:

<img width="399" alt="Screen Shot 2021-08-19 at 4 33 01 PM" src="https://user-images.githubusercontent.com/31013359/130157823-e546a798-f774-4a44-a3ec-d9a5d83ad584.png">

Changes proposed in this PR:

- Add a new block version of the existing banner widget, with the following features:
  - All features of the widget (multiple banner colors, icons, up to two buttons)
  - Ability to hid the 'Close' button to make banners that cannot be dismissed (i.e. the COVID banner)

zainabD reviewed (changes requested):
It looks good! 
I just noticed that the content of the banner is not aligned with the page content, since we're using **container** inside a **container-fluid** for the banner, so it adds padding-x twice. 
I would add a **px-0** to one of the containers to fix the issue: 
![Screen Shot 2021-08-20 at 4 38 16 PM](https://user-images.githubusercontent.com/15880606/130302937-4e7b84e3-c29f-4e8d-990c-95a446a243c0.png)


After adding **px-0** to the container wrapper: 
![Screen Shot 2021-08-20 at 4 37 41 PM](https://user-images.githubusercontent.com/15880606/130302995-e43c83e2-9ea2-40e0-991e-ba79a7f723d0.png)

wmcconnell commented:
Per Zainab's note, I added the `px-0` class to the inner container. This makes the full-width version align the content with the logo, like this:

<img width="700" alt="Screen Shot 2021-08-23 at 3 16 45 PM" src="https://user-images.githubusercontent.com/31013359/130527419-c6c46db3-825f-4581-86d5-06bdf1a2a861.png">

While a banner inside the content area aligns like this:

<img width="613" alt="Screen Shot 2021-08-23 at 3 17 03 PM" src="https://user-images.githubusercontent.com/31013359/130527461-87204c33-7726-4f28-be8f-0fc747ed1bbc.png">

I think that second one is what we would want inside the content area. It would look strange if we aligned the content in that case - as we'd either have no padding inside the banner, or the banner color would "leak" out of the content area.

zainabD reviewed (approved):
Looks good!

wmcconnell commented:
Forgot that I had some ACF field group changes on local that I needed to finish before asking for a review, so I'm re-re-requesting a review.

- Added tabs to split up style and content fields
- Added help text
- Updated the field group description
- Switched to WYSIWYG editor for the title, just in case they need special characters in the future

wmcconnell commented:
I added default text of 'Button One' and 'Button Two' for the buttons, so they will be created as full-sized buttons with labels, like this...

<img width="1012" alt="Screen Shot 2021-08-25 at 9 30 35 AM" src="https://user-images.githubusercontent.com/31013359/130829473-d2966d00-deeb-4767-aac4-89f5776ec708.png">

zainabD reviewed (approved):
All look good!
